phys2bids/phys2bids.py

```diff
@@ -111,89 +111,11 @@ def print_json(outfile, samp_freq, time_offset, ch_name):
     utils.writejson(outfile, summary, indent=4, sort_keys=False)
 
 
-def use_heuristic(heur_file, sub, ses, filename, outdir, record_label=''):
-    """
-    Import and use the heuristic specified by the user to rename the file.
-
-    Parameters
-    ----------
-    heur_file: path
-        Fullpath to heuristic file.
-    sub: str or int
-        Name of subject.
-    ses: str or int or None
-        Name of session.
-    filename: path
-        Name of the input of phys2bids.
-    outdir: str or path
-        Path to the directory that will become the "site" folder
-        ("root" folder of BIDS database).
-    record_label: str
-        Optional label for the "record" entry of BIDS.
-
-    Returns
-    -------
-    heurpath: str or path
-        Returned fullpath to tsv.gz new file (post BIDS formatting).
-
-    Raises
-    ------
-    KeyError
-        if `bids_keys['task']` is empty
-    """
-    utils.check_file_exists(heur_file)
-
-    # Initialise a dictionary of bids_keys that has already "recording"
-    bids_keys = {'sub': '', 'ses': '', 'task': '', 'acq': '', 'ce': '',
-                 'dir': '', 'rec': '', 'run': '', 'recording': record_label}
-
-    # Start filling bids_keys dictionary and path with subject and session
-    if sub.startswith('sub-'):
-        bids_keys['sub'] = sub[4:]
-        fldr = os.path.join(outdir, sub)
-    else:
-        bids_keys['sub'] = sub
-        fldr = os.path.join(outdir, f'sub-{sub}')
-
-    if ses:
-        if ses.startswith('ses-'):
-            bids_keys['ses'] = ses[4:]
-            fldr = os.path.join(fldr, ses)
-        else:
-            bids_keys['ses'] = ses
-            fldr = os.path.join(fldr, f'ses-{ses}')
-
-    # Load heuristic and use it to fill dictionary
-    heur = utils.load_heuristic(heur_file)
-    bids_keys.update(heur.heur(Path(filename).stem))
-
-    # If bids_keys['task'] is still empty, stop the program
-    if not bids_keys['task']:
-        LGR.warning(f'The heuristic {heur_file} could not deal with'
-                    f'{Path(filename).stem}')
-        raise KeyError('No "task" attribute found')
-
-    # Compose name by looping in the bids_keys dictionary
-    # and adding nonempty keys
-    name = ''
-    for key in bids_keys:
-        if bids_keys[key] != '':
-            name = f'{name}{key}-{bids_keys[key]}_'
-
-    # Finish path, create it, add filename, export
-    fldr = os.path.join(fldr, 'func')
-    utils.path_exists_or_make_it(fldr)
-
-    heurpath = os.path.join(fldr, f'{name}physio')
-
-    return heurpath
-
-
 def phys2bids(filename, info=False, indir='.', outdir='.', heur_file=None,
               sub=None, ses=None, chtrig=0, chsel=None, num_timepoints_expected=0,
               tr=1, thr=None, ch_name=[], chplot='', debug=False, quiet=False):
```

Comment:
`num_timepoints_expected` and `tr` are assumed to be lists in the function body. To fix this, you should set these to default to `None`:

```suggestion
              sub=None, ses=None, chtrig=0, chsel=None, num_timepoints_expected=None,
              tr=None, thr=None, ch_name=[], chplot='', debug=False, quiet=False):
```

and the do a check for this below:

```python
if isinstance(num_timepoints_expected, int)
    num_timepoints_expected = [num_timepoints_expected]
if isinstance(tr, int):
    tr = [tr]
```

(Providing mutable objects—like lists—as parameters to a function is bad practice.)

I recommend this because having default values that aren't `None` but not _acting_ on those values is not advisable. That is, you _only_ use `num_timepoints_expected` and `tr` if they're set to something that isn't the default—so the default should be `None`!